packages/material-ui/src/Stack/Stack.js

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {
+  createUnarySpacing,
+  getValue,
+  handleBreakpoints,
+  unstable_extendSxProp as extendSxProp,
+} from '@material-ui/system';
+import { deepmerge } from '@material-ui/utils';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+
+/**
+ * Return an array with the separator React element interspersed between
+ * each React node of the input children.
+ *
+ * > intersperse([1,2,3], 0)
+ * [1,0,2,0,3]
+ */
+function intersperse(children, separator) {
+  const childrenArray = React.Children.toArray(children).filter(Boolean);
+
+  return childrenArray.reduce((output, child, index) => {
+    output.push(child);
+
+    if (index < childrenArray.length - 1) {
+      output.push(React.cloneElement(separator, { key: `separator-${index}` }));
+    }
+
+    return output;
+  }, []);
+}
+
+const getSideFromDirection = (direction) => {
+  switch (direction) {
+    case 'row':
+      return 'Left';
+    case 'row-reverse':
+      return 'Right';
+    case 'column':
+      return 'Top';
+    // case 'column-reverse':
+    default:
+      return 'Bottom';
+  }
+};
```

Comment:
Did you consider an object map? Would it be faster to read, send over the wire and resolve at runtime?

Reply:
Ah, yes way more simple 😄